subprojects/platform-native/src/main/java/org/gradle/nativeplatform/platform/internal/DefaultNativePlatform.java

```diff
@@ -95,11 +95,11 @@ public DefaultNativePlatform withArchitecture(ArchitectureInternal architecture)
 
     private static class HostPlatform extends DefaultNativePlatform {
         HostPlatform() {
-            super("host", DefaultNativePlatform.getCurrentOperatingSystem(), DefaultNativePlatform.getCurrentArchitecture());
+            super("host:" + DefaultNativePlatform.getCurrentArchitecture().getName(), DefaultNativePlatform.getCurrentOperatingSystem(), DefaultNativePlatform.getCurrentArchitecture());
```

Comment:
I used colon `:` instead because `arch` can be `x86_64` which could make the name confusing regarding the division of the underscore. In the end, the name shouldn't matter with the new plugins. I also hold up from adding the operating system name in the name simply because we have no need for it at the moment.